packages/sourcecred/src/core/credGrainView.js

```diff
@@ -136,6 +136,27 @@ export class CredGrainView {
     return new TimeScopedCredGrainView(this, startTimeMs, endTimeMs);
   }
 
+  withTimeScopeFromLookback(
+    effectiveTimestamp: TimestampMs,
+    numIntervalsLookback: number
+  ): TimeScopedCredGrainView {
+    const intervalsBeforeEffective = this._intervals.filter(
+      (interval) => interval.endTimeMs <= effectiveTimestamp
+    );
+
+    if (!numIntervalsLookback)
+      return new TimeScopedCredGrainView(this, 0, effectiveTimestamp);
+
+    if (intervalsBeforeEffective.length <= numIntervalsLookback)
+      return new TimeScopedCredGrainView(this, 0, effectiveTimestamp);
+
+    return new TimeScopedCredGrainView(
+      this,
+      intervalsBeforeEffective[0].startTimeMs,
```

Comment:
Wait, aren't you supposed to use numIntervalsLookback here?

Reply:
Updated per recommendations. Refactored both immediate and balanced policies. Re manual tested.